packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -388,16 +402,21 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
     /**
      * @inheritdoc IAssetManager
      */
-    function initUndelegate(address validator, uint128 shares) external {
+    function undelegate(address validator, uint128 shares) external {
```

Comment:
Why not changing the input to amount? It's better to use share only for internal calculation and internal purposes, imo.

Reply:
Yeah, I agree. What do you think @sm-stack?

Reply:
Yep I also agree, since it will reduce the complexity to calculate share at the frontend.